src/python/grapl_analyzerlib/grapl_analyzerlib/query_gen.py

```diff
@@ -107,23 +121,19 @@ def find_func(q: "Queryable", var_alloc: VarAllocator) -> str:
                 (isinstance(and_filter[0], Eq) or isinstance(and_filter[0], IntEq))
                 and not and_filter[0].negated
             ):
-                from copy import deepcopy
-
+                filter = deepcopy(and_filter[0])
                 if prop_name == "node_key":
-                    filter = deepcopy(and_filter[0])
-                    v = var_alloc.alloc(Eq("node_key", filter.value))
-                    filter.value = v
+                    filter.value = var_alloc.alloc(Eq("node_key", filter.value))
                     return filter.to_filter()
-                if prop_name == "uid":
-                    filter = deepcopy(and_filter[0])
-                    v = var_alloc.alloc(Eq("uid", filter.value))
-                    filter.value = v
+                elif prop_name == "uid":
+                    filter.value = var_alloc.alloc(Eq("uid", filter.value))
                     return filter.to_filter()
+                else:
+                    filter.value = var_alloc.alloc(Eq(filter.predicate, filter.value))
+                    singular_eq_nu = filter.to_filter()
```

Comment:
This is the major fix of this PR - we previously weren't inserting the VarAlloc thing here